src/wallet/api.rs

```diff
@@ -928,7 +928,7 @@ impl Wallet {
             .expect("Internal Keychain expected");
         let addresses = self.rpc.derive_addresses(
             change_branch_descriptor,
-            Some([next_change_addr_index, next_change_addr_index + count]),
+            Some([next_change_addr_index, next_change_addr_index + count - 1]),
         )?;
```

Comment:
Why do we even have `addresses`? Instead of just one next internal address. We only used count=1 in all places because thats the only sane use case of derive address. If we need multiple addresses we can just call this in repeat. The RPC overhead isn't too bad to worry about, and with BDK we will remove RPC dependence.

So my suggestion is, to remove this count arg. Make it look exactly the same as `get_next_external_address`, and we can forget about sanity-checking the count.